### ISSUE TITLE
Price filter gets negative price 

### ISSUE THREAD

**Issue by ioweb-gr** [Smile-SA/elasticsuite#3300]
Every now and then in some categories, the price slider starts to show negative prices in the lowest margin. The upper boundary is correct. 

### Preconditions

Magento Version : 2.4.6-p4 CE

ElasticSuite Version :2.11.5.1 

Environment : Production mode

Third party modules :

### Steps to reproduce

1. I'm not sure how to reproduce this but I keep getting negative prices in the price filter. 

### Expected result

1. The price filter starts from the minimum price in the product listing. 

### Actual result
![image](https://github.com/Smile-SA/elasticsuite/assets/20220341/a45f5c53-fb79-4ff8-9f05-35fcea6de641)

**Comment by vahonc** [comment]
Hello @ioweb-gr,

It's more likely that this problem is related to your environment and its settings. I'm not sure that we can reproduce it without fully understanding how everything is configured. 

Are you sure that there are no established discounts or catalog price rules, in other words, something that could affect on a negative price?

BR,
Vadym

**Comment by ioweb-gr** [comment]
Hi @vahonc We don't actualy use any catalog rules 

![image](https://github.com/Smile-SA/elasticsuite/assets/20220341/4f43bc68-155d-4bb2-b4eb-3a731e121d52)

As far as the filter is concerned it's behaving really weird

Check what happens as I change the price slider 


https://github.com/Smile-SA/elasticsuite/assets/20220341/9d0d0729-0b03-48bb-80f8-2fc2ee9cb664


I'm not really sure where to start to provide the require info. 

Any ideas what I should check and provide the necessary info?

**Comment by vahonc** [comment]
@ioweb-gr,

What is the status of this setting?

![Screenshot from 2024-06-19 15-17-03](https://github.com/Smile-SA/elasticsuite/assets/67791361/8ce045a0-4dfa-4af5-a9ce-043d339067da)

What about your fronted theme? Is there fully compatibility (based on) with default Luma theme? If it possible try to switch to Luma theme.

Have you checked the price of your products or discount rules?

BR,
Vadym

**Comment by ioweb-gr** [comment]
It's set to Yes

![image](https://github.com/Smile-SA/elasticsuite/assets/20220341/baaac969-9b98-469b-973e-e6498b17527b)

I've sampled a lot of them, and they appear proper. Moreover, when I sort them by price Ascending, no product appears with a negative price on the catalog and everything is purchaseable properly at the displayed price.

**Comment by romainruaud** [comment]
can you enable the debug mode of Elasticsuite and give us the query it's generating ?

if you can replay this query via cerebro (if you're familiar with it), I'd like to be able to show the results as well.

Regards

**Comment by ioweb-gr** [comment]
I'm not familiar with cerebro. I'll try to check the debug mode when there's a bit less traffic

**Comment by ioweb-gr** [comment]
> What about your fronted theme? Is there fully compatibility (based on) with default Luma theme? If it possible try to switch to Luma theme.
Forgot to answer this is the LUMA theme, we just changed styles

**Comment by ioweb-gr** [comment]
Just an update here I think we're suffering from this 

https://github.com/magento/magento2/issues/24953
https://github.com/magento/magento2/issues/15609

**Comment by vahonc** [comment]
@ioweb-gr,

Are you able to reproduce this issue without (disabled) ElasticSuite module?

>can you enable the debug mode of Elasticsuite and give us the query it's generating ?

Have you found something that can help us to understand your problem?

BR,
Vadym

**Comment by ioweb-gr** [comment]
Hi 

Yes actually I tested further and the negative prices appear after removing elastic suite as well so it's not the reason the indexes are created wrong but the core itself.

What I haven't been able to determine because I cannot remove it for a prolonged period of time is whether the prices in the configurable products are created because the elastic suite is enabled or not

The culprit however is that configurable products suddenly have prices stored in the db which create a problem in the indexer.

**Comment by rikwillems** [comment]
@romainruaud if we can agree that Magento prices never go below 0 we could do the following to fix the issue in the price filter.

File: src/module-elasticsuite-catalog/Block/Navigation/Renderer/PriceSlider.php line 77
```
-         $minValue = $this->getFilter()->getMinValue();
+        $minValue = max(0, $this->getFilter()->getMinValue());
```

**Comment by romainruaud** [comment]
@rikwillems indeed but this would not really solve the root cause of the issue, which I'd like to know : if we see negative price, it means at some point they get indexed in Elasticsearch, but also in the Magento price index table, which is very strange.

Regards

**Comment by rikwillems** [comment]
@romainruaud that is true. But as the related Magento tickets are quite old already I went down this road. For my project I have this as a patch, but for sure good to track the origin issue.

**Comment by ioweb-gr** [comment]
@rikwillems This doesn't help either because it will give you invalid range. 

Let's say that normally products start at 50$ price and max is 120$

If the filter which was negative now shows 0 as the minimum, the user can set the price filter to minimum 0 and max 40 for example and he'd see 0 products.

It will still be a mistake as he should see the lowest boundary properly

**Comment by romainruaud** [comment]
In any cases, if you have a negative price in `catalog_product_index_price` table, you're f**ed, because we'll just basically assume Magento computed the price properly and we'll index this in ES "as is".

So if this is the case, it's not an Elasticsuite related bug, but a Magento one, and we'll have no way to fix this here :(

regards

**Comment by vahonc** [comment]
@ioweb-gr,

Is this issue still relevant to you?

Perhaps the previous comment will also be useful to you.

BR,
Vadym
